### PR TITLE
Thonny: 3.0.5 -> 3.1.2

### PR DESCRIPTION
**Pull request by leenaars**
###### Motivation for this change

Version bump as per repology of this Python 3 IDE. Upstream project has moved to Github.
Packaged a new dependency (asttokens).

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/asttokens/1.6.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{ lib, fetchPypi, buildPythonPackage, pythonOlder, isPyPy
```
Comment (FRidh) [inline pkgs/development/python-modules/asttokens/1.6.nix]: Wrong file

Comment (leenaars) [reply pkgs/development/python-modules/asttokens/1.6.nix]: Repaired. Apologies!

Comment (ryantm) [reply pkgs/development/python-modules/asttokens/1.6.nix]: Why are you adding a file called 1.6.nix in asttokens that contains a definition for a package called asteroid? Does Thonny depend on asteroid?

**Review by ryantm — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/asttokens/1.6.nix (hunk 1 above)
Comment (leenaars) [reply pkgs/development/python-modules/asttokens/1.6.nix]: No, this was a commit error.

**Review by ryantm — APPROVED** [review]

**Comment by ryantm** [comment]
@GrahamcOfBorg build thonny

**Comment by ryantm** [comment]
runs on my NixOS machine, looks goods.

**Pull request merged**